vars/isBuildFailure.groovy

```diff
@@ -0,0 +1,23 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/**
+  Return true if the build status is FAILURE or UNSTABLE
+*/
+def call() {
+  return currentBuild.result == 'FAILURE' || currentBuild.result == 'UNSTABLE'
```

Comment:
NIT, though we might need to implement the function `resultIsWorseOrEqualTo` if it's not provided by the test library

```suggestion
  return currentBuild.resultIsWorseOrEqualTo('UNSTABLE')
```

Reply:
currentBuild is a variable in the test binding, this is stored in a Map, to allow call this method we have to introduce unnecessary complexity in the tests, so I reverted the change.